### PR TITLE
Data Science with Python: One Hot Encoding

### LINKED ISSUE #1843: Data Science with Python : One Hot Encoding
Welcome to 'DSWP' Team, good to see you here

This issue will helps readers in gaining all the guidance that one needs to know about getting started with One Hot Encoding. Tutorial to An Introduction to One Hot Encoding and how it's applied using sample code

To get assigned to this issue, add your serial numbers mentioned in the spreadsheet of "Data Science with Python", the approach one would follow and choice you prefer (Documentation, Audio, Video). You can go with all three or any number of options you're interested to work on.

If you had referred any resources, add them up in "DS Resources". Similarly if you had used datasets, include them in "DS Datasets".

Domain : Machine Learning

Mentors Assigned : Dhruv Bajaj / Yash Goel

Points to Note :

The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.

"Issue Title" and "PR Title should be the same. Include issue number along with it.

Changes should be made inside the Datascience_With_Python/ directory & Datascience_With_Python branch.

Follow Contributing Guidelines & Code of Conduct before start Contributing.

This issue is only for 'GWOC' contributors of 'DSWP' domain.

All the best. Enjoy your open source journey ahead. 😎

### PR DESCRIPTION
**Pull request by Sakshi1700**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #1843 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [x] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+```bash
+  hot_enc = pd.get_dummies(df2, columns = ['region', 'sex','smoker'])
+  hot_enc
+```
+We passed region, sex and smoker as the value for the prefix attribute of the get_dummies() method, hence you can see the strings region, sex and smoker prefixed before the header of each of the one-hot encoded columns in the output.
+
+Running the above code yeilds.
+
+![Approach_1](https://user-images.githubusercontent.com/56879294/134985137-60809310-c40b-41aa-a433-390bec96b31c.png)
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md:56]: Please remove code images of the notebook from readme. Please do not explain code in the readme, just give explain the concepts and basic introduction.

File: Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+Running the above code yeilds.
+
+![Approach_2](https://user-images.githubusercontent.com/56879294/134985238-121c77ed-579b-465a-a9fc-a65b6421df15.png)
+
+## Dataset and Libraries Used
+
+* Dataset : [Insurance.csv](https://www.kaggle.com/mirichoi0218/insurance)
+
+* Tools used : Pandas and Sklearn
+
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md:75]: Add the python libraries used in the notebook here.

File: Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+## Conclusion 
+* It is important to encode categorical variables to make good predictions and extract information.
+* Works when well the categorical variable is non-ordinal.
+* One issue with one hot encoder is multicollinearity.
+* There are two approaches one using Pandas and another using Sklearn.
+
+
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md:82]: Once changes in the readme are done, I will go through the notebook.

**Review by DhruvBajaj01 — COMMENTED** [review]
Please make the suggested changes in readme

File: Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md (hunk 1 above)
Comment (Sakshi1700) [reply Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md:56]: Should I remove output images or images and code both?

Comment (DhruvBajaj01) [reply Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md:56]: You need not explain code in the readme.

Comment (DhruvBajaj01) [reply Datascience_With_Python/Machine Learning/Tutorials/One Hot Encoding/README.md:56]: Like readme is more about the concept explanation, so no need to add code snip here

**Review by prathimacode-hub — APPROVED** [review]
PR approved. 👍@Sakshi1700

**Pull request merged**